src/formatDate.js

```diff
@@ -50,7 +50,54 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  let year = '';
+  let month = '';
+  let day = '';
+
+  for (const i of fromFormat) {
+    if (i === 'YY' || i === 'YYYY') {
+      year += dateArr[fromFormat.indexOf(i)];
+    }
+
+    if (i === 'MM') {
+      month += dateArr[fromFormat.indexOf(i)];
+    }
+
+    if (i === 'DD') {
+      day += dateArr[fromFormat.indexOf(i)];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && year.length < 4) {
+    if (year < 30) {
+      year = '20' + year;
+    } else {
+      year = '19' + year;
+    }
+  }
+
+  if (toFormat.includes('YY') && year.length > 2) {
+    year = year.split('').slice(2).join('');
+  }
+
+  const dateFormated = [];
+
+  for (const i of toFormat) {
```

Comment:
Use a more descriptive variable name, instead of 'i' use 'element' or 'formatElement'.